pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -45,6 +45,26 @@ self: super: {
   # Needs older QuickCheck version
   attoparsec-varword = dontCheck super.attoparsec-varword;
 
+  # Fixed at head, but hasn't cut a release in awhile.
+  darcs = doJailbreak (overrideCabal super.darcs (drv: {
+    patches = [
+      # cabal 3.0 support
+      ( pkgs.fetchpatch {
+        url = "https://raw.githubusercontent.com/input-output-hk/haskell.nix/9f8a6e3e5755e4b917686c6267b1135caf724f1f/patches/darcs-setup.patch";
+        sha256 = "sha256-LJDc22EUHchm7hzVVs6LvG115T6OBAI8BdmsGK38GdE=";
+      })
+    ];
+    postPatch = (drv.postPatch or "") + ''
+      sed -i -e 's/< *0.5/<1/' -e 's/< *2.13/<3/' -e 's/< *4.13/<5/' -e 's/< *1.3/<2/' darcs.cabal
+      sed -i '/sDistHook = /,+14d' Setup.hs
+      sed -i '/fail   = failSM/i instance MonadFail SM where' src/Darcs/Patch/ReadMonads.hs
+      sed -i 's/fail \$/error $/' src/Darcs/Util/Tree/Monad.hs src/Darcs/Repository/Diff.hs src/Darcs/Patch/Prim/V1/Apply.hs
+      sed -i 's/Monad m => PatchInfoAnd/MonadFail m => PatchInfoAnd/' src/Darcs/Patch/PatchInfoAnd.hs
+      sed -i 's/fail /error /' src/Darcs/Patch/Depends.hs src/Darcs/Patch/Match.hs src/Darcs/Repository/Match.hs
+      sed -i '/fail _ /i instance  MonadFail Perhaps where' src/Darcs/Patch/{,Prim}/V1/Commute.hs
+    '';
```

Comment:
This is pretty hard to understand what is going on.  Is there any way to get a patch for this instead of using `sed`?

Reply:
Yes, please provide a patch for those changes. A patch fails with a recognizable error message once the new version of darcs comes out, whereas these commands might produce strange compiler failures 10 minutes into the build for all we know.